### DISCUSSION TITLE
Does the minio shard upload method uploadpart() not support multithreading?

### DISCUSSION THREAD

**Discussion by flyerzhou** [minio/minio#19686] [Q&A]
Recently, minio was used for file storage in the project. The large files were sharded on the backend and multi-threaded. Testing found that under multi-threaded conditions, the time consumption is no different from serial. Is it possible that mini does not support multi-threaded mode or there are other areas that need to be set? The following is a code snippet. Please help take a look. The minio  version used is 8.3.3,Here is a screenshot of the code
<img width="857" alt="1715044011409" src="https://github.com/minio/minio/assets/38522410/82ca0a74-fe4c-47aa-bf77-21d64383ef11">
<img width="808" alt="1715043681018" src="https://github.com/minio/minio/assets/38522410/284c106b-fc38-45fd-82a0-cb1cedfc96e9">

**Comment by klauspost** [comment]
It depends on the client. The MinIO server handles concurrent part uploads fine. The [go sdk](https://github.com/minio/minio-go) supports this when appropriate as an example.
